node/pkg/db/governor.go

```diff
@@ -176,12 +190,35 @@ func unmarshalOldTransfer(data []byte) (*Transfer, error) {
 	}
 	t.EmitterAddress = emitterAddress
 
-	msgID := make([]byte, 256)
-	n, err := reader.Read(msgID)
-	if err != nil || n == 0 {
-		return nil, fmt.Errorf("failed to read vaa id [%d]: %w", n, err)
+	msgIdLen := uint16(0)
+	if err := binary.Read(reader, binary.BigEndian, &msgIdLen); err != nil {
+		return nil, fmt.Errorf("failed to read msgID length: %w", err)
 	}
-	t.MsgID = string(msgID[:n])
+
+	if msgIdLen > 0 {
+		msgID := make([]byte, msgIdLen)
+		n, err := reader.Read(msgID)
+		if err != nil || n == 0 {
+			return nil, fmt.Errorf("failed to read vaa id [%d]: %w", n, err)
+		}
+		t.MsgID = string(msgID[:n])
+	}
+
+	hashLen := uint16(0)
+	if err := binary.Read(reader, binary.BigEndian, &hashLen); err != nil {
+		return nil, fmt.Errorf("failed to read hash length: %w", err)
+	}
+
+	if hashLen > 0 {
+		hash := make([]byte, hashLen)
+		n, err := reader.Read(hash)
+		if err != nil || n == 0 {
+			return nil, fmt.Errorf("failed to read hash [%d]: %w", n, err)
+		}
+		t.Hash = string(hash[:n])
+	}
```

Comment:
Is there a straightforward way to add tests for these error conditions? It would be nice to have error condition coverage for this feature if it isn't too onerous of a request.

Reply:
Added negative tests for the unmarshal functions.